src/FINDOLOGIC/Api/Responses/Xml21/Xml21Response.php

```diff
@@ -68,26 +63,38 @@ protected function buildResponseElementInstances($response)
         $this->promotion = $this->getPromotionFromResponse($xmlResponse);
         $this->results = new Results($xmlResponse->results[0]);
 
-        foreach ($xmlResponse->products->children() as $product) {
-            $productId = ResponseHelper::getStringProperty($product->attributes(), 'id', true);
-            // Set product ids as keys for the products.
-            $this->products[$productId] = new Product($product);
+        try {
```

Comment:
Having just a try/catch catching all exceptions is not a clean solution.
Check if `$xmlResponse->products` and `$xmlResponse->products->children()` exist and are not empty. If they have a value and are not empty, do the foreach.